libvast/vast/system/eraser.hpp

```diff
@@ -0,0 +1,84 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include <string>
+
+#include <caf/fwd.hpp>
+
+#include "vast/fwd.hpp"
+#include "vast/ids.hpp"
+#include "vast/system/query_processor.hpp"
+
+namespace vast::system {
+
+/// Periodically queries the INDEX with a configurable expression and erases
+/// all hits from the ARCHIVE.
+class eraser_state : public query_processor {
+public:
+  // -- member types -----------------------------------------------------------
+
+  using super = query_processor;
+
+  // -- constants --------------------------------------------------------------
+
+  static inline constexpr const char* name = "eraser";
+
+  // -- constructors, destructors, and assignment operators --------------------
+
+  eraser_state(caf::event_based_actor* self);
+
+  void init(caf::timespan interval, std::string query, caf::actor index_hdl,
+            caf::actor archive_hdl);
```

Comment:
Along with my other comment, I would find `index` and `archive` more natural names for the actor variables. Put differently, the handle aspect should be conveyed by the type, not the variable name.